bundle/deploy/terraform/tfdyn/convert_job.go

```diff
@@ -19,8 +20,34 @@ func convertJobResource(ctx context.Context, vin dyn.Value) (dyn.Value, error) {
 		log.Debugf(ctx, "job normalization diagnostic: %s", diag.Summary)
 	}
 
+	// Sort the tasks of each job in the bundle by task key. Sorting
+	// the task keys ensures that the diff computed by terraform is correct and avoids
+	// recreates. For more details see the NOTE at
+	// https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/job#example-usage
+	// and https://github.com/databricks/terraform-provider-databricks/issues/4011
+	vout := vin
+	var err error
+	tasks, ok := vin.Get("tasks").AsSequence()
+	if ok {
+		sort.Slice(tasks, func(i, j int) bool {
+			tk1, ok := tasks[i].Get("task_key").AsString()
+			if !ok {
+				return true
+			}
+			tk2, ok := tasks[j].Get("task_key").AsString()
+			if !ok {
+				return false
+			}
```

Comment:
What is the significance of the `!ok` branches?

If it doesn't matter then they can be ignored.

Reply:
That's pretty much what we do here. The `!ok` checks are there to return early if the task key is not defined in the task by mistake, since the final validation for task_key is performed on the server side. 

As for the return values, I simply treat the absence of `task_key` as lower in the lexographical ordering for consistency. Added a comment saying as much.